src/fastga_he/models/propulsion/components/loads/sm_pmsm/components/perf_iron_losses.py

```diff
@@ -0,0 +1,194 @@
+# This file is part of FAST-OAD_CS23-HE : A framework for rapid Overall Aircraft Design of Hybrid
+# Electric Aircraft.
+# Copyright (C) 2025 ISAE-SUPAERO
+
+import numpy as np
+import openmdao.api as om
+
+
+class PerformancesIronLosses(om.ExplicitComponent):
+    """
+    Computation of the iron losses of the SM PMSM results from design air gap magnetic flux density, obtained from the
+    least square surrogate model (II-67) in :cite:`touhami:2020`. The default value is obtained
+    from :cite:`pyrhonen:2013`.
+    """
+
+    def initialize(self):
+        self.options.declare(
+            name="motor_id", default=None, desc="Identifier of the motor", allow_none=False
+        )
+        self.options.declare(
+            "number_of_points", default=1, desc="number of equilibrium to be treated"
+        )
+
+    def setup(self):
+        motor_id = self.options["motor_id"]
+        number_of_points = self.options["number_of_points"]
+
+        self.add_input(
+            "electrical_frequency",
+            units="Hz",
+            val=np.nan,
+            shape=number_of_points,
+            desc="The oscillation frequency of the SM PMSM AC current",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:"
+            + motor_id
+            + ":design_air_gap_flux_density",
+            val=0.9,
+            units="T",
+            desc="The design air gap magnetic flux density",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":mass",
+            val=np.nan,
+            units="kg",
+        )
+
+        self.add_output(
+            "iron_power_losses",
+            units="kW",
+            val=0.0,
+            shape=number_of_points,
+            desc="Iron losses of the SM PMSM due to altering magnetic flux",
+        )
+
+    def setup_partials(self):
+        motor_id = self.options["motor_id"]
+        number_of_points = self.options["number_of_points"]
+
+        self.declare_partials(
+            of="iron_power_losses",
+            wrt=["electrical_frequency"],
+            method="exact",
+            rows=np.arange(number_of_points),
+            cols=np.arange(number_of_points),
+        )
+        self.declare_partials(
+            of="iron_power_losses",
+            wrt=[
+                "data:propulsion:he_power_train:SM_PMSM:"
+                + motor_id
+                + ":design_air_gap_flux_density",
+                "data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":mass",
+            ],
+            method="exact",
+            rows=np.arange(number_of_points),
+            cols=np.zeros(number_of_points),
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        motor_id = self.options["motor_id"]
+
+        mass = inputs["data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":mass"]
+        air_gap_flux_density = inputs[
```

Comment:
I'ùm just grasping it now, but the iron losses during the mission depend on the design point ?